src/components/Album/Album.js

```diff
@@ -20,12 +20,11 @@ function Album() {
         setLoading(false);
       })
       .catch(error => {
-        console.error('Error fetching album details:', error);
+        handleApiError(error);
       })
-      .finally(() => {
-        setLoading(false);
-      });
-  }, [id, currentUser]);
+      
+      return () => setLoading(false);
+  }, [id]);
 
   const handleAddToCollections = () => {
```

Comment:
Noticing a console error when multiple albums with the same masterId are added saying key should be unique, do we want to add some logic here so they can only be added once?

Reply:
I also noticed that from Josh's branch. I added this in the descriptions of the sad path ticket to just get things moving.